p4-fusion/main.cc

```diff
@@ -249,6 +249,23 @@ int Main(int argc, char** argv)
 	ThreadPool::GetSingleton()->Initialize(networkThreads);
 	SUCCESS("Created " << ThreadPool::GetSingleton()->GetThreadCount() << " threads in thread pool");
 
+	auto t = std::thread([]()
```

Comment:
So to confirm, our error handling strategy is to have a "watcher" thread that blocks on waiting for the thread pool to shutdown. One it shuts down, it'll catch any exceptions that were re-thrown by the threadpool, and then exit the main process.

Reply:
Yes, that's the idea! Maybe this is a bit too much of a Go pattern, but it seemed to work in practice, at least